theano/tensor/nnet/neighbours.py

```diff
@@ -111,6 +111,43 @@ def grad(self, inp, grads):
                 return [neibs2images(gz, neib_shape, x.shape, mode=self.mode),
                         grad_undefined(self, 1, neib_shape),
                         grad_undefined(self, 2, neib_step)]
+
+        if self.mode in ['valid']:
```

Comment:
Good. Scan will be slower then implementing it natively, but it is better to have it implemented with scan, then not having it!